### PR TITLE
Feature/save coordinates 2 tz

### PR DESCRIPTION
**Pull request by theresazobel**
#2 Saves coordinates of bounding boxes in csv file when "save" button is clicked. 
Format:
"Img Name" - Upper-left X - Upper-left Y - Lower-right X - Lower-right Y
(might be changed depending on what SSD needs)

### CODE REVIEW

**Comment by fapaul** [comment]
Looks good so far but I had some problem until I saw where my image ends and the background begins especially for images with transparent background. I would suggest highlighting the boundaries of the image. 
I also can not see the "save" button, do you mean the next button? Where are the bounding boxes saved?

File: src/app/app.component.ts [hunk 1]
```diff
@@ -56,19 +58,33 @@ export class AppComponent implements OnInit {
...
     };
+    this.imgName = this.buildings[this.index]['name'];
     reader.readAsDataURL(this.buildings[this.index]);
   }
 
   public resetImage() {
     this.processCanvas();
   }
 
+  public saveToCsv() {
```
Comment (fapaul) [inline src/app/app.component.ts]: Currently, only public functions are used, could we convert the one just used within the class to private?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -19,6 +19,8 @@ export class AppComponent implements OnInit {
   public index = 0;
   public imgAmount;
   public msg;
+  public boundingBoxes = [];
+  public imgName = '';
 
   constructor() { }
```
Comment (fapaul) [inline src/app/app.component.ts]: Is the constructor function needed in this case?

**Comment by theresazobel** [comment]
@mockingbird2 I will add the border - good idea.
Regarding the save button, it looks like this for me:

screen shot 2017-11-12 at 23 06 52

Does it look different for you? Can you post a screenshot 👍

File: src/styles.css [hunk 3]
```diff
@@ -11,4 +11,8 @@
 
 #msg {
     margin-bottom: 2%;
+}
+
+#imageCanvas {
+    border: 2px solid black;
 }
```
Comment (sfriedo) [inline src/styles.css]: Oh, Oh, Oh! No newline at EOF! not sure if acceptable... 🌵

File: src/app/app.component.ts [hunk 4]
```diff
@@ -56,19 +58,33 @@ export class AppComponent implements OnInit {
...
+    this.imgName = this.buildings[this.index]['name'];
     reader.readAsDataURL(this.buildings[this.index]);
   }
 
   public resetImage() {
     this.processCanvas();
   }
 
+  public saveToCsv() {
+    let csvContent = 'data:text/csv;charset=utf-8,';
```
Comment (sfriedo) [inline src/app/app.component.ts]: Am I missing something or doesn't the csv include a header? It seems to me you're just writing the values. 

(If this is intended behaviour, fine for me, just wanted to point it out)

Comment (theresazobel) [reply src/app/app.component.ts]: yes its just the values. I wasn't sure if we need headers. If so, then let me know & i will add them

Comment (sfriedo) [reply src/app/app.component.ts]: I'd prefer it so we don't have to assume (hardcode) it when parsing the files.

Comment (theresazobel) [reply src/app/app.component.ts]: Okay then how would you call them?

Img - Upper-left X - Upper-left Y - Lower-right X - Lower-right Y ?

Comment (sfriedo) [reply src/app/app.component.ts]: Yes.

Comment (theresazobel) [reply src/app/app.component.ts]: done

**Review by fapaul — APPROVED** [review]
LGTM

**Pull request merged**